dashboard/app/controllers/admin_users_controller.rb

```diff
@@ -100,4 +100,39 @@ def manual_pass
     flash[:alert] = "UserLevel (ID: #{user_level.id}) updated"
     redirect_to :manual_pass_form
   end
+
+  def permissions_form
+  end
+
+  # Grants the indicated permission to the indicated user. Expects params[:email] and
+  # params[:permission] to be populated.
+  def grant_permission
```

Comment:
It would be worth extracting the core functionality in this action to a method on the `User` model.